optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +246,153 @@ public void skipAhead() throws ExecutionException, InterruptedException {
+                            entity.setValue(solution.getValueList().get(0));
+                            scoreDirector.afterVariableChanged(entity, "value");
+                            scoreDirector.triggerVariableListeners();
+                        }), new CustomPhaseConfig().withCustomPhaseCommands(
+                        (ScoreDirector<TestdataSolution> scoreDirector) -> {
+                            TestdataSolution solution = scoreDirector.getWorkingSolution();
+                            TestdataEntity entity = solution.getEntityList().get(1);
+                            scoreDirector.beforeVariableChanged(entity, "value");
+                            entity.setValue(solution.getValueList().get(1));
+                            scoreDirector.afterVariableChanged(entity, "value");
+                            scoreDirector.triggerVariableListeners();
+                        }));
+
+        SolverManagerConfig solverManagerConfig = new SolverManagerConfig().withParallelSolverCount(String.valueOf(processCount));
+
+        return SolverManager.create(solverConfig, solverManagerConfig);
+    }
+
+    private void assertSolveWithoutConsumer(int processCount, SolverManager<TestdataSolution, Integer> solverManager) throws InterruptedException, ExecutionException {
+        List<SolverJob<TestdataSolution, Integer>> jobs = new ArrayList<>(processCount);
+
+        for (int id = 0; id < processCount; id++) {
+            jobs.add(solverManager.solve(id, PlannerTestUtils.generateTestdataSolution("s" + id, 2)));
+        }
+
+        for (SolverJob<TestdataSolution, Integer> job : jobs) {
+            assertSolutionInitialized(job.getFinalBestSolution());
+        }
+    }
+
+    private void assertSolveWithBestSolutionConsumer(int processCount, SolverManager<TestdataSolution, Integer> solverManager)
+            throws InterruptedException {
+        int bestSolutionEventTriggeredCountExpected = processCount * 2;
+        CountDownLatch bestSolutionConsumerLatch = new CountDownLatch(bestSolutionEventTriggeredCountExpected);
+
+        AtomicInteger bestSolutionEventTriggerCount = new AtomicInteger();
+        Consumer<TestdataSolution> bestSolutionConsumer = bestSolution -> {
+            bestSolutionEventTriggerCount.incrementAndGet();
+            bestSolutionConsumerLatch.countDown();
+        };
+        for (int id = 0; id < processCount; id++) {
+            solverManager.solveAndListen(
+                    id,
+                    problemId -> PlannerTestUtils.generateTestdataSolution("s" + problemId, 2),
+                    bestSolutionConsumer, null);
+        }
+        bestSolutionConsumerLatch.await();
+
+        assertThat(bestSolutionEventTriggerCount.intValue()).isSameAs(bestSolutionEventTriggeredCountExpected);
+    }
+
+    private void assertSolveWithFinalBestSolutionConsumer(int processCount, SolverManager<TestdataSolution, Integer> solverManager)
```

Comment:
It might be interesting to also check if the best solution publish to the consumer is the same as the best solution available via the `SolverJob`.

Reply:
Done for the consumer that gets the finalBestSolution. As for the solveAndListen portion, I am not entirely sure how to obtain the bestSolution that is not final from a job. Currently I check whether the final ones are contained in the list of consumers and that the list is of size 2x the problem count. Any ideas how to deal with this issue?

Reply:
You can have (and probably should have) a separate consumer for each submitter problem. That's how the API most likely will be used. In such a case, knowing there are two best solutions per problem, just take the second one.